### DISCUSSION TITLE
Form Text Filed Fires OnChange events even if the value hasnt changed.

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#7499] [Help] [answered]
Server : Latest self hosted, Debian, 
Client : Mozilla Firefox, MacOSX Monterrey, M1

It seems that the Form Text Field fires its OnChange event sequence on OnBlur, even if the value hasn't changed.
So focus / unfocus the control will fire the events again.

Shouldn't the event fire only when there is an actual change in the value or should the event be renamed to "OnBlur" rather than "onChange" ?

**Comment by R2bEEaton** [comment, marked as answer]
I personally find it useful that it triggers after each time it loses focus, but one way to get your desired effect is by using the OnChange action to check if some App State is equal to the form field. If it's not, then update the state. You can then use conditional actions after that to perform your desired task.

**Reply by poirazis** [reply]
Hey @R2bEEaton , its cool that it fires onBlur and you can use a Continue if Action to check if the value has changed or not before proceeding. But the naming is misleading. Maybe there should be more Events for each widget, onBlur and onChange and onEnter and onKeypress etc ...

**Reply by R2bEEaton** [reply]
That's fair, you make a good point. Would definitely love more options almost everywhere on almost everything.

**Reply by melohagan** [reply]
Hey @poirazis 

You're absolutely spot on, the naming isn't quite right for how it behaves, but as Ryan pointed out it's easy to workaround.

Nonetheless I've raised a bug here: https://github.com/Budibase/budibase/issues/7533
